### PR TITLE
Ut3252/porting to python3

### PR DESCRIPTION
**Pull request by baptisteguidoux**

### CODE REVIEW

**Comment by jfboismenu** [comment]
Hi David, does this mean the the Unity integration will only run with Python 3 now? importlib is Python 3 only. Toolkit and all our apps and frameworks will remain compatible with Python 2.7 in addition to Python 3.7 for the foreseable future.

I’m asking at some point the Unity integration relied on the Shotgun Desktop’s Python. Is that going to change?

**Comment by lassond** [comment]
Hi @jfboismenu !
Only Windows is supported with the unity engine right now, with precise instructions on how to setup Python with PySide. We tested on the Mac with the Shotgun Desktop's Python interpreter but never released anything.

File: engine.py [hunk 1]
```diff
@@ -113,7 +113,7 @@ def post_app_init(self):
         
         # store the menu cmds so that we can access them later from C#
         self._menu_cmd_items = {}
-        for (cmd_name, cmd_details) in self.commands.items():
+        for (cmd_name, cmd_details) in list(self.commands.items()):
```
Comment (benoithudson) [inline engine.py:116]: why can't you iterate over the items?

Comment (benoithudson) [reply engine.py:116]: same question with the other two times you make a list below

Comment (baptisteguidoux) [reply engine.py:116]: Sorry about that, I used 2to3 to help me, it gave me warning messages about `items` not returning a list, but yes it's still iterable, and I guess better memory and time wise.

**Pull request merged**